python/tests/test_tables.py

```diff
@@ -3540,6 +3540,18 @@ def test_table_references(self):
         assert str(populations) == before_populations
         assert str(provenances) == before_provenances
 
+    def test_compute_mutation_parents_ignores_existing_values(self):
+        tables = tskit.TableCollection(sequence_length=1.0)
+        parent = tables.nodes.add_row(time=1.0)
+        child = tables.nodes.add_row(time=0, flags=tskit.NODE_IS_SAMPLE)
+        tables.edges.add_row(left=0.0, right=1.0, parent=parent, child=child)
+        site = tables.sites.add_row(position=0.0, ancestral_state="A")
+        tables.mutations.add_row(site=site, node=child, derived_state="C")
+        tables.build_index()
+        tables.mutations.parent[:] = 42
+        tables.compute_mutation_parents()
+        assert tables.mutations.parent[0] == tskit.NULL
+
```

Comment:
Another test checking the parent data is preserved on error

Reply:
if possible we should check the data is preserved on error even if `TSK_NO_CHECK_INTEGRITY` is on